mono/mini/mini-llvm.c

```diff
@@ -5980,9 +5980,6 @@ process_bb (EmitContext *ctx, MonoBasicBlock *bb)
 				 * FIXME: If later code uses the regs defined by these instructions,
 				 * compilation will fail.
 				 */
-				const char *spec = INS_INFO (next->opcode);
-				if (spec [MONO_INST_DEST] == 'i')
-					ctx->values [next->dreg] = LLVMConstNull (LLVMInt32Type ());
 				MONO_DELETE_INS (bb, next);
```

Comment:
Does it make sense to delete those instructions at all? I mean LLVM should be able to delete them itself, isn't it?

Reply:
The problem is they are unreachable so they can reference unitialized vregs, all kinds of mess.